os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -133,6 +139,39 @@ void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size
 	for (int idx = startidx; idx <= endidx; idx++) {
 		heapinfo_parse_heap(&heap[idx], HEAPINFO_DETAIL_ALL, HEAPINFO_PID_ALL);
 	}
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	if (heap_type != KERNEL_HEAP) {
+		mfdbg("#####################################################################################\n\n");
+		mfdbg("*************************************************************************************\n");
+		mfdbg("                          Summary of Other heap memory usage                         \n");
+		mfdbg("*************************************************************************************\n\n");
+		struct mm_heap_s *kheap = kmm_get_baseheap();
+		mfdbg("*************************************************************************************\n");
+		mfdbg("                          Summary of Kernel heap memory usage                        \n");
+		mfdbg("*************************************************************************************\n\n");
+		for (int idx = HEAP_START_IDX; idx <= HEAP_END_IDX; idx++) {
+			heapinfo_parse_heap(&kheap[idx], HEAPINFO_SIMPLE, HEAPINFO_PID_ALL);
+		}
+
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
```

Comment:
We need to print all app heap irrespective of common binary. So, you will need to move this config check below to line 164 and also add an else part to print the app heap if common binary is not present.